testutils/clustermanager/gke.go

```diff
@@ -207,15 +214,21 @@ func (gs *GKEClient) Setup(r GKERequest) ClusterOperations {
 	return gc
 }
 
-// Initialize checks environment for cluster and projects to decide whether using
+// initialize checks environment for cluster and projects to decide whether using
 // existing cluster/project or creating new ones.
-func (gc *GKECluster) Initialize() error {
+func (gc *GKECluster) initialize() error {
 	// Try obtain project name via `kubectl`, `gcloud`
 	if gc.Project == nil {
 		if err := gc.checkEnvironment(); err != nil {
 			return fmt.Errorf("failed checking existing cluster: '%v'", err)
-		} else if gc.Cluster != nil { // return if Cluster was already set by kubeconfig
-			return nil
+		} else if gc.Cluster != nil { // Return if Cluster was already set by kubeconfig
+			// If clustername provided is different from kubeconfig, don't use it
+			if gc.Request != nil && gc.Request.ClusterName != "" && gc.Cluster.Name != gc.Request.ClusterName {
```

Comment:
I'm not sure I understand the rationale here... :/

Reply:
The use case in my mind is that for local runs, I want the cluster to be created with name `foo`,  and my local kubeconfig is pointing at `bar` cluster, by passing `ClusterName = foo` as argument I would expect not to use the cluster pointed by my kubeconfig, but create a new cluster for me. Open to discussion though

Reply:
Makes sense. The comment isn't clear to me to illustrate this scenario, left a suggestion, hopefully it makes sense to you.

Reply:
Your suggestion makes sense to me, submitted it